src/System.Private.ServiceModel/src/System/ServiceModel/Channels/SslStreamSecurityUpgradeProvider.cs

```diff
@@ -610,11 +610,6 @@ protected override Stream OnInitiateUpgrade(Stream stream, out SecurityMessagePr
                 throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new SecurityNegotiationException(
                     SR.Format(SR.NegotiationFailedIO, ioException.Message), ioException));
             }
-            catch (Exception exception)
-            {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new SecurityNegotiationException(
```

Comment:
Dang -- I'll need to find a way to distinguish those.  The custom validator test code also throws a vanilla Exception because that is how it is done in the WCF samples.  I'll explore HResults a bit to see if I can use the to decide when to wrap.